frontend/src/components/SearchBar.js

```diff
@@ -1,7 +1,35 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { PropTypes } from 'prop-types';
 
+
+const debounce = (func, wait) => {
+  let timeout;
+
+  return function executedFunction(...args) {
+    const later = () => {
+      clearTimeout(timeout);
+      func(...args);
+    };
+
+    clearTimeout(timeout);
+    timeout = setTimeout(later, wait);
+  };
+};
+
 function SearchBar(props) {
+  const [inputVal, setInputVal] = useState('');
+  const [debounceFunc, setFunc] = useState(null);
+
+  useEffect(() => {
+    const debFn = debounce(props.updateBySearch, 500);
```

Comment:
I'm not familiar with debounce. What does the 500 refer to?

Reply:
That's a good point, I might want to implement a sort of global variable to get rid of the hard-coded number there. The 500 stands for the number of milliseconds the search bar (based on the debouncer) is going to wait for the user to stop typing. After that amount of time, the debouncer is going to call the function updateBySearch.